### PR TITLE
Fix using ? as a parameter

### LINKED ISSUE #202: doesn't support sql parameter in set clause
When use parser to parse the sql like this:
```sql
  update tbl_customer t set t.`description`=:yp where t.id=1;
```
the parser failed.it seems the parser cann't handle sql parameter in the SET sub-clause.

### LINKED ISSUE #412: Support for parameterized select clauses
When using placeholders in a select clause for a prepared statement, the parser stops at the first placeholder.  For example, parsing "SELECT 1, 2, ?, 3" will include the expressions 1 and 2 in the SelectStatement object, but not the placeholder or 3.
```
PhpMyAdmin\SqlParser\Statements\SelectStatement Object
(
    [expr] => Array
        (
            [0] => PhpMyAdmin\SqlParser\Components\Expression Object
                (
                    [database] => 
                    [table] => 
                    [column] => 
                    [expr] => 1
                    [alias] => 
                    [function] => 
                    [subquery] => 
                )

            [1] => PhpMyAdmin\SqlParser\Components\Expression Object
                (
                    [database] => 
                    [table] => 
                    [column] => 
                    [expr] => 2
                    [alias] => 
                    [function] => 
                    [subquery] => 
                )

        )

    [from] => Array
        (
        )

    [index_hints] => 
    [partition] => 
    [where] => 
    [group] => 
    [having] => 
    [order] => 
    [limit] => 
    [procedure] => 
    [into] => 
    [join] => 
    [union] => Array
        (
        )

    [end_options] => 
    [options] => PhpMyAdmin\SqlParser\Components\OptionsArray Object
        (
            [options] => Array
                (
                )

        )

    [first] => 0
    [last] => 11
)
```

### LINKED ISSUE #492: Missing support for SET = ?
The following query results in an error with the message of `Missing expression.` when its parsed. 
 
```sql
-- gives error
UPDATE  order
SET     price = ?
WHERE   orderid = ?
```

```sql
-- does not error
UPDATE  order
SET     price = 3
WHERE   orderid = ?
```

### PR DESCRIPTION
**Pull request by MauricioFauth**
- Fixes #202
- Fixes #412
- Fixes #492
- Closes #499
- Related to #172

### CODE REVIEW

File: tests/data/misc/parseParameter2.out [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+                67,
+                0
+            ],
+            [
+                "Variable name was expected.",
+                "?",
+                70,
+                0
+            ]
+        ],
```
Comment (williamdes) [inline tests/data/misc/parseParameter2.out:363]: should this still exist ?

Comment (MauricioFauth) [reply tests/data/misc/parseParameter2.out:363]: You right. That's another issue. Thanks!

File: tests/data/bugs/gh492.out [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+                0
+            ],
+            [
+                "Unrecognized keyword.",
+                {
+                    "@type": "@4"
+                },
+                0
+            ]
+        ]
```
Comment (williamdes) [inline tests/data/bugs/gh492.out:258]: I think this is the only remaining bit to fix

**Review by williamdes — APPROVED** [review]
Awesome !
Will it be easy to detect the ? or := for our (broken) feature "Bind parameters" in the query box ?

File: tests/data/bugs/gh492.out (hunk 2 above)
Comment (MauricioFauth) [reply tests/data/bugs/gh492.out:258]: This should not be removed because the statement is actually wrong. The table name should be quoted.

Comment (williamdes) [reply tests/data/bugs/gh492.out:258]: oh okay, that's why

**Comment by MauricioFauth** [comment]
> Will it be easy to detect the ? or := for our (broken) feature "Bind parameters" in the query box ?

Is there an issue for that?

**Comment by williamdes** [comment]
> > Will it be easy to detect the ? or := for our (broken) feature "Bind parameters" in the query box ?
> 
> Is there an issue for that?

No, I did not report it until now since the parser has unhappy

**Pull request merged**